modelzoo/MMoE/train.py

```diff
@@ -0,0 +1,673 @@
+                                                   combiner='mean'))
+        elif column_name in IDENTITY_INPUTS:
+            column = tf.feature_column.categorical_column_with_identity(column_name, 50)
+            feature_cols.append(
+                tf.feature_column.embedding_column(column,
+                                                   dimension=16,
+                                                   combiner='mean'))
+        else:
+            raise ValueError('Unexpected column name occured')
+
+    return feature_cols
+
+'''
+END OF INPUT CONFIG SPECIFICS
+'''
+'''
+MODEL CONFIG SPECIFICS
+'''
+DNN_ACTIVATION = tf.nn.relu
+
+L2_REGULARIZATION = 1e-06
+EMBEDDING_REGULARIZATION = 5e-05
+
+EXPERTS_COUNT = 4
+EXPERT_HIDDEN_UNITS = [256, 192, 128, 64]
+
+#Tower tuple structure (tower name, label name, hidden units)
+TOWERS = [
+    ("ctr", "clk", [256, 192, 128, 64]),
+    ("cvr", "buy", [256, 192, 128, 64])
+]
+'''
+MODEL CONFIG SPECIFICS
+'''
+def make_scope(name, bf16, part):
+    if(bf16):
+        return tf.variable_scope(name, partitioner=part, reuse=tf.AUTO_REUSE).keep_weights()
+    else:
+        return tf.variable_scope(name, partitioner=part, reuse=tf.AUTO_REUSE)
+
+def add_layer_summary(value, tag):
+    tf.summary.scalar('%s/fraction_of_zero_values' % tag,
+                      tf.nn.zero_fraction(value))
+    tf.summary.histogram('%s/activation' % tag, value)
+
+def l2_regularizer(scale, scope=None):
+  if isinstance(scale, numbers.Integral):
+    raise ValueError(f'Scale cannot be an integer: {scale}')
+  if isinstance(scale, numbers.Real):
+    if scale < 0.:
+      raise ValueError(f'Setting a scale less than 0 on a regularizer: {scale}.')
+    if scale == 0.:
+      return lambda _: None
+
+  def l2(weights):
+    with tf.name_scope(scope, 'l2_regularizer', [weights]) as name:
+      my_scale = tf.convert_to_tensor(scale, dtype=weights.dtype.base_dtype, name='scale')
+      return tf.math.multiply(my_scale, tf.nn.l2_loss(weights), name=name)
+
+  return l2
+
+class MMOE():
+    def __init__(self,
+                 input,
+                 feature_column,
+                 num_experts,
+                 expert_hidden_units,
+                 towers,
+                 l2_scale,
+                 learning_rate=0.1,
+                 bf16=False,
+                 input_layer_partitioner=None,
+                 dense_layer_partitioner=None,
+                 is_training=True):
+        self.__feature_column = feature_column
+        self.__num_experts = num_experts
+        self.__expert_hidden_units = expert_hidden_units
+        self.__towers = towers
+        self.__l2_regularization = l2_regularizer(l2_scale) if l2_scale else None
+        self.__learning_rate = learning_rate
+        self.__bf16 = bf16
+        self.__input_layer_partitioner = input_layer_partitioner
+        self.__dense_layer_partitioner = dense_layer_partitioner
+        self.__is_training = is_training
```

Comment:
the variable "self.__is_training" is not used in the code.

Reply:
Removed self.__is_training.